### PR TITLE
TRAINING-10: Create contributors guide

### PR DESCRIPTION
**Pull request by soenkeliebau**

### CODE REVIEW

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 1]
```diff
@@ -20,84 +20,58 @@
...
 - Promoting the project (articles, blog posts, talks at conferences)
-- Documentation
+- Writing documentation
 
-=== Pull-Requests
+=== Finding a Project
+If you want to get involved, but do not have a specific project in mind please feel free to check our issue tracker for https://issues.apache.org/Jira/issues/?jql=project%20%3D%20TRAINING%20AND%20labels%20%3D%20beginner%20AND%20assignee%20in%20(EMPTY)[unassigned issues with the _beginner_ label]. These should be good candidates to get your feet wet and gather some experience before tackling larger issues.
 
-The simplest way to submit code changes is via a GitHub pull-request.
+If nothing catches your eye and you come up with something that you want to work on by yourself, please do share your intent by either posting on the mailing list or creating an issue, as this allows us to limit duplication of effort as much as possible.
```
Comment (dspavlov) [inline site/src/site/asciidoc/developers/contributing.adoc]: I suggest encouraging usage of the list. Every active contributor reads the list, but JIRA issue may be lost and disappeared from the radar

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 2]
```diff
@@ -20,84 +20,58 @@
...
 
-    git clone https://github.com/{your-user-id}/incubator-training.git
+For reviews we distinguish between _content_ commits that add content to training decks and _functional_ commits like code changes to tools or similar things.
 
-Now you have a copy of Apache Training on your computer and you can change whatever you want and as it's your copy, you can even commit these changes without any danger of breaking things.
+*Code Commits* require one approving review.
+This review has to be done by a committer if the pull request was created by a non-committer. If a committer creates a pull request and explicitly requests a review from a non-committer, then this is review is sufficient.
 
-As soon as you're finished with your changes and want us to have a look, it's time to create a so-called `Pull-Request`.
+*Content Commits* require one or more approving reviews. As we do not expect to initially have committers that are also experts in all fields for which we host content there may be a need for two reviews in this case: one for form by a committer and one for content by someone who considers himself an SME.
```
Comment (dspavlov) [inline site/src/site/asciidoc/developers/contributing.adoc]: I suggest using 'himself or herself' as well as it is in the following sentence

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 3]
```diff
@@ -20,84 +20,58 @@
...
-(If no remote is provided, git will use `origin` per default)
+*Trivial Changes*
+Committers can, at their discretion, decide that a change is trivial and does not need to be explicitly approved.
+In this case, the pull request or Jira issue should be marked accordingly and left open for review for at least 72 hours.
+After this period has passed it can be merged without review.
 
 ===  Bug Reports
 
-We use https://issues.apache.org/jira/projects/TRAINING[JIRA] as our Bug &amp; Issue Tracker.
+We use https://issues.apache.org/Jira/projects/TRAINING[JIRA] as our Issue Tracker.
```
Comment (dspavlov) [inline site/src/site/asciidoc/developers/contributing.adoc]: URL with 'Jira' here (from the capital letter) not found I suggest a change to jira

Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: f*** - that was search and replace stupidity. Will fix!

File: site/src/site/asciidoc/developers/contributing.adoc (hunk 2 above)
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: Will fix

File: site/src/site/asciidoc/developers/contributing.adoc (hunk 1 above)
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: I'm not sure about this tbh. If there is a discussion about something specific that ends up as a commit, personally I'd prefer the meat of that discussion to be had on Jira.
I always thought that mail threads have a tendency to become extremely hard to follow for various reasons.

Also, new Jiras are reported to the mailing list by Jira, so anybody following the list should see those, no?

Comment (lfrancke) [reply site/src/site/asciidoc/developers/contributing.adoc]: On the other hand jiras are easier to search and reference

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 4]
```diff
@@ -20,84 +20,58 @@
...
+If you want to get involved, but do not have a specific project in mind please feel free to check our issue tracker for https://issues.apache.org/Jira/issues/?jql=project%20%3D%20TRAINING%20AND%20labels%20%3D%20beginner%20AND%20assignee%20in%20(EMPTY)[unassigned issues with the _beginner_ label]. These should be good candidates to get your feet wet and gather some experience before tackling larger issues.
 
-The simplest way to submit code changes is via a GitHub pull-request.
+If nothing catches your eye and you come up with something that you want to work on by yourself, please do share your intent by either posting on the mailing list or creating an issue, as this allows us to limit duplication of effort as much as possible.
 
-In order to do this first create a GitHub account and sign into your account.
+=== Code Contributions
+Many of the actions listed above will take the form of contributions to the git repository, for which we have defined a few basic guidelines to ensure we can all collaborate effectively.
+Code contributions can take two forms, you can either create a pull request on github or directly attach patch files to a Jira issue for review.
```
Comment (lfrancke) [inline site/src/site/asciidoc/developers/contributing.adoc]:
```suggestion
Code contributions can take two forms, you can either create a pull request on GitHub or directly attach patch files to a Jira issue for review.
```

File: site/src/site/asciidoc/developers/contributing.adoc (hunk 1 above)
Comment (dspavlov) [reply site/src/site/asciidoc/developers/contributing.adoc]: I'm not insisting on that. Another issue in a bigger community that no-one actually knows what is in JIRA. Without reminding/discussion issue using dev.list it is almost impossible to get attention to the problem. But I have to admit I'm trying to solve problems that not happened yet. So it is perfectly ok to keep both options

Comment (justinmclean) [reply site/src/site/asciidoc/developers/contributing.adoc]: Emails are searchable (either using your email client) or lists.apache.org. Using lists.apache.org you can also create permalinks to an email.

File: site/src/site/asciidoc/developers/contributing.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 5]
```diff
@@ -20,84 +20,58 @@
...
 
-After that's done, please go to our https://github.com/apache/incubator-training[GitHub site] and create a so-called `Fork`.
+A Jira ticket and a pull request are not mutually exclusive, in fact, we encourage people to create both, first a Jira to discuss the issue, task, improvement in general and later on a pull request for actual implementation discussions.
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 6]
```diff
@@ -20,84 +20,58 @@
...
+If you follow this process, please prefix the name of your pull request with the Jira issue number, this will ensure that they automatically get linked.
+If, for example, you open a pull request to fix issue TRAINING-123, then your PR title could read "_TRAINING-123: Fix stuff_".
 
-image::contributing-github-fork.png[]
+If you are not familiar with git, how to create pull requests and similar workflows, below is a list of useful links that should help you get started.
+But as with all other things, please do reach out to us on the mailing list if anything is unclear and you need help, we are more than happy to help!
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 7]
```diff
@@ -20,84 +20,58 @@
...
-Now ideally you check-out your cloned repository:
+==== Review Process
+The Training project follows a review-then-commit workflow, which basically means that no commit should be made to the repository that has not been reviewed and approved by the required number of persons.
 
-    git clone https://github.com/{your-user-id}/incubator-training.git
+For reviews we distinguish between _content_ commits that add content to training decks and _functional_ commits like code changes to tools or similar things.
 
-Now you have a copy of Apache Training on your computer and you can change whatever you want and as it's your copy, you can even commit these changes without any danger of breaking things.
+*Code Commits* require one approving review.
+This review has to be done by a committer if the pull request was created by a non-committer. If a committer creates a pull request and explicitly requests a review from a non-committer, then this is review is sufficient.
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

File: site/src/site/asciidoc/developers/contributing.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: I've incorporated this, but to be quite honest am unsure if the way I did it is correct. I tried googling and _think_ this may be right, but could you please have a quick look?

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 8]
```diff
@@ -20,84 +20,58 @@
...
 
-As soon as you're finished with your changes and want us to have a look, it's time to create a so-called `Pull-Request`.
+*Content Commits* require one or more approving reviews. As we do not expect to initially have committers that are also experts in all fields for which we host content there may be a need for two reviews in this case: one for form by a committer and one for content by someone who considers himself an SME.
+If the committer considers him- or herself to be an expert on the material one review is sufficient, just like for code commits.
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: see above

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 9]
```diff
@@ -20,84 +20,58 @@
...
-
-    git push
-
-(If no remote is provided, git will use `origin` per default)
+*Trivial Changes*
+Committers can, at their discretion, decide that a change is trivial and does not need to be explicitly approved.
+In this case, the pull request or Jira issue should be marked accordingly and left open for review for at least 72 hours.
+After this period has passed it can be merged without review.
 
 ===  Bug Reports
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

File: site/src/site/asciidoc/developers/contributing.adoc [hunk 10]
```diff
@@ -20,84 +20,58 @@
 
 === Forms of contribution
 
-There are multiple forms in which you can become involved with the Apache Training project.
+The Apache Training (Incubating) project welcomes contributions in all forms, sizes, and shapes.
+Below you can find a small list of possible ways that you can contribute if you have some spare time and want to get involved.
 
-These usually are, but are not limited to:
+This is by no means a finite list, there are without doubt dozens of other useful ways that you can pick, please do not let this limit your creativity!
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: done

File: site/src/site/asciidoc/developers/contributing.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (justinmclean) [reply site/src/site/asciidoc/developers/contributing.adoc]: Still says "himself"?

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply site/src/site/asciidoc/developers/contributing.adoc]: Is that by any chance browser cache or a github link directly to that commit?
I just checked and for me it reads "themselves" when I go to the "files changed" link ..

[the remark this answers is not in the document]
Comment (justinmclean) [reply site/src/site/asciidoc/developers/contributing.adoc]: Odd in one place it's right here it's not. Revised copy looks good to me.
